deploy/helm/grafana-operator/templates/service.yaml

```diff
@@ -8,8 +8,8 @@ spec:
   type: {{ .Values.kubeRbacProxy.service.type }}
   ports:
     - port: {{ .Values.kubeRbacProxy.service.port }}
-      targetPort: http
+      targetPort: metrics
       protocol: TCP
-      name: http
+      name: https
```

Comment:
But it isn't https traffic, it's http.
I would argue that you should revert to http, else I think it looks good.

Or am I thinking wrong?

Reply:
@NissesSenap thanks for the quick review!!!!

not https, are you sure????

this is my current service monitor that uses https. I used the helm chart to deploy with no modifications

<details><summary>Commands to demonstrate SSL</summary>

the service monitor using https to connect.

``` bash
k8s-757d55d849-d22wl:/apps# kubectl -n operators describe servicemonitor grafana-operator
Name:         grafana-operator
Namespace:    operators
Labels:       app.kubernetes.io/component=controller
              app.kubernetes.io/managed-by=Helm
              app.kubernetes.io/name=grafana-operator
              app.kubernetes.io/part-of=operators
              app.kubernetes.io/version=0.63.0
Annotations:  meta.helm.sh/release-name: tester
              meta.helm.sh/release-namespace: default
API Version:  monitoring.coreos.com/v1
Kind:         ServiceMonitor
Metadata:
  Creation Timestamp:  2023-03-28T04:09:42Z
  Generation:          22
  Managed Fields:
    API Version:  monitoring.coreos.com/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:meta.helm.sh/release-name:
          f:meta.helm.sh/release-namespace:
        f:labels:
          .:
          f:app.kubernetes.io/component:
          f:app.kubernetes.io/managed-by:
          f:app.kubernetes.io/name:
          f:app.kubernetes.io/part-of:
          f:app.kubernetes.io/version:
      f:spec:
        .:
        f:endpoints:
        f:selector:
    Manager:         helm
    Operation:       Update
    Time:            2023-03-28T07:57:29Z
  Resource Version:  31124
  UID:               71ef8485-a32d-4454-a5d0-97a9fa05ce31
Spec:
  Endpoints:
    Bearer Token File:  /var/run/secrets/kubernetes.io/serviceaccount/token
    Honor Labels:       true
    Interval:           10s
    Port:               https
    Scheme:             https
    Tls Config:
      Insecure Skip Verify:  true
  Selector:
    Match Labels:
      app.kubernetes.io/name:  grafana-operator
Events:                        <none>
k8s-757d55d849-d22wl:/apps#
```

here is a wget showing issues with certificate validation and a bad connection when protocol not https

``` bash
/prometheus $ wget http://grafana-operator-metrics-service.operators.svc:8443/metrics -O- -S
Connecting to grafana-operator-metrics-service.operators.svc:8443 (10.96.84.61:8443)
  HTTP/1.0 400 Bad Request
wget: server returned error: HTTP/1.0 400 Bad Request
/prometheus $
/prometheus $
/prometheus $ wget https://grafana-operator-metrics-service.operators.svc:8443/metrics -O- -S
Connecting to grafana-operator-metrics-service.operators.svc:8443 (10.96.84.61:8443)
wget: note: TLS certificate validation not implemented
  HTTP/1.1 401 Unauthorized
wget: server returned error: HTTP/1.1 401 Unauthorized
/prometheus $
/prometheus $
/prometheus $ wget https://grafana-operator-metrics-service.operators.svc:8443/metrics -O- -S --no-check-certificate
Connecting to grafana-operator-metrics-service.operators.svc:8443 (10.96.84.61:8443)
  HTTP/1.1 401 Unauthorized
wget: server returned error: HTTP/1.1 401 Unauthorized
/prometheus $
```

here is the certificate from grafana-operator pod, collected with openssl

``` bash
k8s-757d55d849-d22wl:/apps# echo -n | openssl s_client -connect grafana-operator-metrics-service.operators.svc:8443
CONNECTED(00000003)
depth=1 CN = grafana-operator-5455c89596-wmj6c-ca@1679989587
verify error:num=19:self-signed certificate in certificate chain
verify return:1
depth=1 CN = grafana-operator-5455c89596-wmj6c-ca@1679989587
verify return:1
depth=0 CN = grafana-operator-5455c89596-wmj6c@1679989587
verify return:1
---
Certificate chain
 0 s:CN = grafana-operator-5455c89596-wmj6c@1679989587
   i:CN = grafana-operator-5455c89596-wmj6c-ca@1679989587
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Mar 28 06:46:26 2023 GMT; NotAfter: Mar 27 06:46:26 2024 GMT
 1 s:CN = grafana-operator-5455c89596-wmj6c-ca@1679989587
   i:CN = grafana-operator-5455c89596-wmj6c-ca@1679989587
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Mar 28 06:46:26 2023 GMT; NotAfter: Mar 27 06:46:26 2024 GMT
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIDTzCCAjegAwIBAgIBAjANBgkqhkiG9w0BAQsFADA6MTgwNgYDVQQDDC9ncmFm
YW5hLW9wZXJhdG9yLTU0NTVjODk1OTYtd21qNmMtY2FAMTY3OTk4OTU4NzAeFw0y
MzAzMjgwNjQ2MjZaFw0yNDAzMjcwNjQ2MjZaMDcxNTAzBgNVBAMMLGdyYWZhbmEt
b3BlcmF0b3ItNTQ1NWM4OTU5Ni13bWo2Y0AxNjc5OTg5NTg3MIIBIjANBgkqhkiG
9w0BAQEFAAOCAQ8AMIIBCgKCAQEAv8MjUOkkCIhD0MfWx7N0W97XMgmkJyH0VBn4
rnWo+1V6Dgtdc1+1fvwb4MjYUkPnbKL1jCKDs4EF08DhKV41mQneMs10hCjN7QAZ
KgucwRLbVo3LQlXq/cm15i+QxKCGzGVULAIRc1B6Q0lG12qfTeOlY4BReA+L52AV
RtIfS+fmhtBCSRQAoPt19oXiB5HmspQ5nb/b6dHHJyl/9apNKZnmOjGMB2mywndh
Ci7rZjQBV6VThROfgzUTBPY+QlCYREe5dLnfJ7Nw/MrA7JpT+fAqDXEMgwBHoXwm
7Lhhz66WC4s31/xewaxoaPQ77kSs0TlQwWRqPFwkukViKhf0WQIDAQABo2MwYTAO
BgNVHQ8BAf8EBAMCBaAwEwYDVR0lBAwwCgYIKwYBBQUHAwEwDAYDVR0TAQH/BAIw
ADAsBgNVHREEJTAjgiFncmFmYW5hLW9wZXJhdG9yLTU0NTVjODk1OTYtd21qNmMw
DQYJKoZIhvcNAQELBQADggEBAHgMEX1uSniTrVyUqmRdezVZB3WukuPLGG2XE+pN
rsPgmStRhzIK6kAqOBJxcmr7yKq63VA9SdmhlJU6wqBRwgUJHANDAwWZYYWhuJis
MTVhls8HFbJXa8CETFISDQ+mW8dWz5Krqut5pdJsSuW/qn/WAhUQvY45yUkmE3oA
gpqisC0KQ+3IuKBbyTs9IFuU7WqsW/DTnCZG17k37FSNFWKgnoijQr+Tn2rGbuu6
RjkV5lz7HF5aROKYkRBZl8DD6g7R9Y6k+ilrpeQ5N+tXw8QXgSumuJxdwzDuKESW
R2Akj/T7PyQPXhjrs1GrwHdrhO4FtOXbDJvVSCIurhNx3lo=
-----END CERTIFICATE-----
subject=CN = grafana-operator-5455c89596-wmj6c@1679989587
issuer=CN = grafana-operator-5455c89596-wmj6c-ca@1679989587
---
No client certificate CA names sent
Peer signing digest: SHA256
Peer signature type: RSA-PSS
Server Temp Key: X25519, 253 bits
---
SSL handshake has read 2186 bytes and written 416 bytes
Verification error: self-signed certificate in certificate chain
---
New, TLSv1.3, Cipher is TLS_AES_128_GCM_SHA256
Server public key is 2048 bit
Secure Renegotiation IS NOT supported
No ALPN negotiated
Early data was not sent
Verify return code: 19 (self-signed certificate in certificate chain)
---
---
Post-Handshake New Session Ticket arrived:
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : TLS_AES_128_GCM_SHA256
    Session-ID: F60B29E936E45827FC51B2406F3C7968BD3D611B37A9E2692E5D096FCF21BF0B
    Session-ID-ctx:
    Resumption PSK: C118DA7E40DA9A331B588C12456D0E465ECA11F571F7A4F6EAECE3DA2EC431DB
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 604800 (seconds)
    TLS session ticket:
    0000 - a5 4f 85 a2 54 27 77 58-75 22 84 07 69 d7 89 39   .O..T'wXu"..i..9
    0010 - 6f bf 45 92 a9 26 94 d6-69 3a 8b 99 a1 67 ad c5   o.E..&..i:...g..
    0020 - 83 a8 fa d2 18 d3 4b 61-c6 60 3d 29 b7 f8 ca 2f   ......Ka.`=).../
    0030 - e0 a9 d7 22 83 d4 10 60-61 de 7e ec 97 01 ac cc   ..."...`a.~.....
    0040 - c4 d6 ea 15 c5 e1 c3 2b-8b 9f 6f 3c 7a ee 5c d3   .......+..o<z.\.
    0050 - d1 97 21 59 b4 16 cd b0-99 4d 2d 4b 25 4e c0 12   ..!Y.....M-K%N..
    0060 - 70 60 72 fc 35 21 60 b7-e0 cf f2 43 41 e6 cb 9b   p`r.5!`....CA...
    0070 - 93                                                .

    Start Time: 1679999468
    Timeout   : 7200 (sec)
    Verify return code: 19 (self-signed certificate in certificate chain)
    Extended master secret: no
    Max Early Data: 0
---
read R BLOCK
DONE
k8s-757d55d849-d22wl:/apps#
```
</p>
</details>

Reply:
on further investigation `kube-rbac-proxy` is setup for secure listen in the deployment config

``` yaml
- name: kube-rbac-proxy
          image: gcr.io/kubebuilder/kube-rbac-proxy:v0.8.0
          imagePullPolicy: IfNotPresent
          args:
            - --secure-listen-address=0.0.0.0:8443
            - --upstream=http://127.0.0.1:8080/
            - --logtostderr=true
            - --v=10
          ports:
          - containerPort: 8443
            name: https
            protocol: TCP
```

Reply:
Hi @jon-nfc, thanks allot for your very detailed description.

My question just showed how little I have used metrics from the operator so far ^^
Thanks allot for showing it, as you say it looks very much like https traffic. I will merge the PR